### PR TITLE
Add reporting for user flows

### PR DESCRIPTION
**Pull request by JasonPapanicholasWF**
## Description

- Add aggregate and HTML reporting for user flows
- Changes to support shared functions between profile and flow

## Type of Change

- [ ] Bug Fix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](https://github.com/wayfair-incubator/oss-template/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: flows/example-test.js [hunk 1]
```diff
@@ -5,23 +5,19 @@ class exampleUserFlow {
     // return new Promise(async (resolve, reject) => {
     const page = await browser.newPage();
```
Comment (spember) [inline flows/example-test.js]: I'm curious, how will this flow file be used? Is this a sample that you're using for testing? Will end users have to write their own, and if so, how does it get injected into the racer?

File: packages/race-cli/src/index.ts [hunk 2]
```diff
@@ -136,13 +137,24 @@ program
...
+  .option(
+    '--output-target <string>',
+    descriptionHelper('Location to save results'),
+    'results'
+  )
+  .option(
+    '--output-format [string...]',
+    descriptionHelper('Save results as CSV, HTML, and/or MD')
+  )
   .usage('./my-test.js')
```
Comment (spember) [inline packages/race-cli/src/index.ts:149]: just noticed this: .js test output?

File: packages/race-cli/src/reporters/aggregate-console-reporter.ts [hunk 3]
```diff
@@ -1,44 +1,57 @@
...
 import logger from '../logger';
 import {formatFilename} from '../helpers';
 
 const STD_DEVIATION_KEY = 'Standard Deviation';
 const MEAN_KEY = 'Mean';
-const METRIC_KEY = 'Metric';
+const METRIC_KEY = 'Measurement';
 const FORMAT_MD = 'md';
 
 const resultsToMarkdown = (
```
Comment (spember) [inline packages/race-cli/src/reporters/aggregate-console-reporter.ts:35]: beyond the scope of this PR but something to keep in mind that we might want to break out the markdown, html, json, console, etc "formats" out into a separate 'output formats' file. just a thought

File: packages/race-cli/src/reporters/aggregate-console-reporter.ts [hunk 4]
```diff
@@ -79,20 +112,79 @@ export class AggregateConsoleReporter extends BaseRacepointReporter {
...
+
+      Object.values(LightHouseAuditKeys).forEach((value) => {
+        if (step.lhr.audits[value]) {
+          activeStep[value].push(step.lhr.audits[value].numericValue);
+        }
+      });
     });
   };
 
   async finalize(cacheStats?: CacheMetricData): Promise<void> {
```
Comment (spember) [inline packages/race-cli/src/reporters/aggregate-console-reporter.ts:160]: this function is bordering on "too big", imo.

I like the Math Operation objects, but you could, for example, extract these two out into a 'default MathBook' (also, great name)

File: packages/race-cli/src/reporters/html-reporter.ts [hunk 5]
```diff
@@ -15,16 +15,16 @@ export class HtmlReporter extends BaseRacepointReporter {
     this._reportPath = outputTarget || '.';
   }
 
-  process = (results: LighthouseResultsWrapper): undefined => {
+  process = (results: UserFlowResultsWrapper): undefined => {
     const htmlPath = `${this._reportPath}/${formatFilename({
-      url: results.lhr.requestedUrl,
-      date: results.lhr.fetchTime,
+      url: results.steps[0].lhr.requestedUrl,
```
Comment (spember) [inline packages/race-cli/src/reporters/html-reporter.ts:20]: is the html reporter always going to be concerned with only the first step, or will there be others?

**Review by spember — APPROVED** [review]
lgtm just minor comments

File: packages/race-cli/src/index.ts (hunk 2 above)
Comment (JasonPapanicholasWF) [reply packages/race-cli/src/index.ts:149]: Correct!

File: packages/race-cli/src/reporters/html-reporter.ts (hunk 5 above)
Comment (JasonPapanicholasWF) [reply packages/race-cli/src/reporters/html-reporter.ts:20]: Its only looking for the URL which is the same on all reports

**Pull request merged**